packages/admin/admin/src/saveBoundary/SaveBoundary.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from "react";
+import { useIntl } from "react-intl";
+import useConstant from "use-constant";
+import { v4 as uuid } from "uuid";
+
+import { messages } from "../messages";
+import { RouterPrompt } from "../router/Prompt";
+
+export type SaveActionSuccess = boolean;
+export interface SaveBoundaryApi {
+    save: () => Promise<SaveActionSuccess>;
+    register: (id: string, props: SaveableProps) => void;
+    unregister: (id: string) => void;
+}
+export interface Saveable {
+    hasErrors: boolean;
+    hasChanges: boolean;
+    saving: boolean;
+}
+
+export const SaveBoundaryApiContext = React.createContext<SaveBoundaryApi | undefined>(undefined);
+export function useSaveBoundaryApi() {
+    return React.useContext(SaveBoundaryApiContext);
+}
+
+export const SaveableContext = React.createContext<Saveable | undefined>(undefined);
```

Comment:
I'm sorry, but it's "savable" (without "e" before "able") 😅 

https://www.oxfordreference.com/display/10.1093/acref/9780197599020.001.0001/acref-9780197599020-e-7969#:~:text=savable%3B%20salvable%3B%20salvageable.&text=Savable%20(so%20spelled%E2%80%94not%20%E2%8B%86,requires%20a%20subscription%20or%20purchase.